### ISSUE TITLE
ValueError: unknown url type: None&output=47

### ISSUE THREAD

**Issue by n1c0la** [rg3/youtube-dl#6913]
$ youtube-dl --verbose http://www.presadiretta.rai.it/dl/portali/site/puntata/ContentItem-3ed19d13-26c2-46ff-a551-b10828262f1b.html
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'--verbose', u'http://www.presadiretta.rai.it/dl/portali/site/puntata/ContentItem-3ed19d13-26c2-46ff-a551-b10828262f1b.html']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.09.09
[debug] Python version 2.7.10 - Darwin-12.6.0-x86_64-i386-64bit
[debug] exe versions: avconv 11.4, avprobe 11.4, ffmpeg 2.8, ffprobe 2.8, rtmpdump 2.4
[debug] Proxy map: {}
[Rai] 3ed19d13-26c2-46ff-a551-b10828262f1b: Downloading webpage
[Rai] 3ed19d13-26c2-46ff-a551-b10828262f1b: Downloading webpage
[Rai] 3ed19d13-26c2-46ff-a551-b10828262f1b: Downloading JSON metadata
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 410, in main
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 400, in _real_main
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1658, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 660, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 287, in extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/rai.py", line 101, in _real_extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 470, in _download_json
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 438, in _download_webpage
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 345, in _download_webpage_handle
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 326, in _request_webpage
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1865, in urlopen
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 423, in open
    protocol = req.get_type()
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 285, in get_type
    raise ValueError, "unknown url type: %s" % self.__original
ValueError: unknown url type: None&output=47

**Comment by danog** [comment]
The presa diretta site is using (yet) another iframe.
Nearly figured it out..

**Comment by danog** [comment]
I figured out how to do this in my bash implementation (/danog/video-dl), but unfortunately I don't know how to rewrite it in Python, I'll try to explain this so that someone can rewrite it if I don't figure it out first: 
```
http://www.rai.it/dl/RaiTV/programmi/media/"$(echo "$file" | sed '/content="ContentItem/!d;s/.*content="//g;s/".*//g')".html

$(echo "$file" | sed '/drawMediaRaiTV/!d;s/.*http/http/g;s/'"'"'.*//g')"

http://www.rai.tv$(echo "$file" | grep -A1 '<div id="idFramePlayer">' | sed '/\<iframe/!d;s/.*src="//g;s/?.*//g')
```
The above code is what I use to get the source video from iframe pages in my bash script: the first and the second method are already implemented, the third isn't. The file variable is the webpage.
Basically, all we need is another iframe Regex match that returns a relative (to rai.tv) Url: note that the only iframe that leads to the video page is the one after the idFramePlayer div.

**Comment by yan12125** [comment]
Rai is fixed (f1388739002a7fd1e8e9c41b642734786fc6c391) and will be available in the next version. Thanks for the report and sorry for the late response.
